spec/models/post_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+
+  describe 'default post details' do
+    let(:user) { create :user }
+    let(:post) { create :post }
+    let(:valid_post) { build :post }
+    let(:comment) { create :comment}
+
+    context "Post validation" do
+      it 'should be valid' do
+        expect(valid_post.valid?).to be true
+      end
+
+      it 'should be invalid without body' do
+        valid_post.body = nil
+        expect(valid_post.valid?).to be false  
+      end
+
+      it 'should be invalid without user' do
+        valid_post.user = nil
+        expect(valid_post.valid?).to be false  
+      end
+    end 
+
+    context "Psost associations" do
+      it 'can have many comments' do
+        post.comments.create(body: "comment 1", user: user)
+        post.comments.create(body: "comment 2", user: user)
+        expect(post.comments.count).to eql(2)
+      end
+      it 'should have any or many likes' do
+        post.likes.create(user: user)
+        expect(post.likes.count).to be >= 0 
+      end 
+
+      it 'should increase likes by one when it is liked by user' do
```

Comment:
Probably the test of increasing and decreasing likes count is better placed in the like_spec